src/HDAA_fmt_plug.c

```diff
@@ -224,8 +245,10 @@ static char *split(char *ciphertext, int index, struct fmt_main *self)
 	cp = strchr(cp+1, '$'); if (!cp) return ciphertext;
 	// now if we have $binary_hash$ then we remove the last '$' char
 	if (strlen(cp) == 1 + BINARY_SIZE*2 + 1) {
-		static char out[256];
-		strnzcpy(out, ciphertext, sizeof(out));
+		static char *out;
+		if (!out)
+			out = mem_alloc_tiny(MAX_CIPHERTEXT_LEN, MEM_ALIGN_NONE);
+		strnzcpy(out, ciphertext, MAX_CIPHERTEXT_LEN);
 		out[strlen(out)-1] = 0;
```

Comment:
I think you need to allocate `MAX_CIPHERTEXT_LEN + 1` and pass that to `strnzcpy`, so that `MAX_CIPHERTEXT_LEN` is treated literally as the maximum length, which is also how you treat ti in the check in `valid`.

Also, drop the unneeded `out[strlen(out)-1] = 0;`

Reply:
`out[strlen(out)-1] = 0;` is needed: it removes the last char (`$`).

But I need `+1` for the length. Otherwise with input of max length, `strnzcpy` would remove the last char itself and then the next line would remove one more char.